python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,380 @@
+"""Torch Module for PGExplainer"""
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import Adam
+
+__all__ = ["PGExplainer"]
+
+
+class PGExplainer(nn.Module):
+    r"""PGExplainer from `Parameterized Explainer for Graph Neural Network
+    <https://arxiv.org/pdf/2011.04573>`
+
+    PGExplainer adopts a deep neural network to parameterize the generation
+    process of explanations, which enables it to explain multiple instances
+    collectively. PGExplainer models the underlying structure as edge
+    distributions, from which the explanatory graph is sampled.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat, embed, edge_weight)`.
+        * The output of its forward function is the logits if embed=False else
+        return the graph embedding.
+    num_features : int
+        Number of input feature for the explanation network.
+    epochs : int, optional
+        Number of epochs to train the explanation network. Default: 10.
+    learning_rate : float, optional
+        Learning rate to train the explanation network. Default: 0.01.
+    coff_size : float, optional
+        Size regularization to constrain the explanation size. Default: 0.01.
+    coff_ent : float, optional
+        Entropy regularization to constrain the connectivity of explanation. Default: 5e-4.
+    init_tmp : float, optional
+        The temperature at the first epoch. Default: 5.0.
+    final_tmp : float, optional
+        The temperature at the final epoch. Default: 1.0.
+    sample_bias : float, optional
+        Some members of a population are systematically more likely to be selected
+        in a sample than others. Default: 0.0.
+    """
+
+    def __init__(
+        self,
+        model,
+        num_features,
+        epochs=10,
+        learning_rate=0.01,
+        coff_size=0.01,
+        coff_ent=5e-4,
+        init_tmp=5.0,
+        final_tmp=1.0,
+        sample_bias=0.0,
+    ):
+        super(PGExplainer, self).__init__()
+
+        self.model = model
+        self.num_features = num_features * 2
+
+        # training parameters for PGExplainer
+        self.epochs = epochs
+        self.learning_rate = learning_rate
+        self.coff_size = coff_size
+        self.coff_ent = coff_ent
+        self.init_tmp = init_tmp
+        self.final_tmp = final_tmp
+        self.sample_bias = sample_bias
+
+        self.init_bias = 0.0
+
+        # Explanation model in PGExplainer
```

Comment:
explanation network?

Reply:
Yes, it is the explanation network that we train which learns the edge distribution.

Reply:
Update the comment for consistency.